### DISCUSSION TITLE
yfinance documentation

### DISCUSSION THREAD

**Discussion by DanielBBar-On** [ranaroussi/yfinance#1330] [General]
Hello all,
Is there documentation for yfinance?
Specifically somewhere where I can see the different fields returned by the API calls and the definition of each one?

Thanks.

**Comment by ValueRaider** [comment]
Not high-quality like the bigger modules. The important methods have docstrings in the source code. Community working on [Wiki page](https://github.com/ranaroussi/yfinance/wiki)

**Comment by ValueRaider** [comment]
I've recently experimented with `sphinx` package + github pages to auto-generate documentation. It generates and displays but I haven't figured out adding custom pages e.g. example use. Links for inspiration:
- webpage with one doctstring (I'm lazy): https://valueraider.github.io/yfinance-cacheyfinance_cache.html#yfinance_cache.yfc_ticker.Ticker.history
- script: https://github.com/ValueRaider/yfinance-cache/blob/docs-html/make-docs.sh

@ericpien

**Reply by ericpien** [reply]
here's a draft: https://ericpien.github.io/yfinance based on: 
- source: https://github.com/ericpien/yfinance/tree/docs/docs
- build: https://github.com/ericpien/yfinance/tree/gh-pages/docs

Let me know the initial feedback. Couple of notes:

- I took a lot of inspiration from: https://pandas.pydata.org/docs/index.html
- I can enhance the docstring on Sector and Industry module in the next iteration
- I left some sample code from README out of this iteration since I want to make sure it's on the right path first.

**Reply by ValueRaider** [reply]
This looks good. @ranaroussi, can we redo the Github Page like this? (only just realised yfinance already had one)

**Reply by ericpien** [reply]
I made further changes:
- Documentation is better structured and covers more classes/methods now. I think it's at a publish-ready state. (https://ericpien.github.io/yfinance/reference/index.html)
- `README` is simplified and references the documentation. (https://github.com/ericpien/yfinance/blob/docs/README.md)
- Push to `docs` branch on my fork builds documentation automatically via GitHub Actions (https://github.com/ericpien/yfinance/actions). This can be modified to work upon merge to `dev` or `main`. (https://github.com/ericpien/yfinance/blob/docs/.github/workflows/deploy_docs.yml)

Let me know what you guys think.

**Reply by ValueRaider** [reply]
Content is good.

Merge strategy has to be targeting the existing `documentation` branch as I cannot change Github Page settings. It has a `docs` folder. But first I need to solve it being 3 years old.

**Reply by ericpien** [reply]
Great. I made the change to build and deploy from the `documentation` branch's `docs` folder. i.e. my fork's [/documentation/docs](https://github.com/ericpien/yfinance/tree/documentation/docs) by adding `publish_branch: documentation` on [line 34 of deploy_doc.yml](https://github.com/ericpien/yfinance/blob/40e93881cfd274871f3dbcb36eb9463e7c5a8311/.github/workflows/deploy_doc.yml#L34) and `destination_dir: docs` on [line 36 of deploy_doc.yml](https://github.com/ericpien/yfinance/blob/40e93881cfd274871f3dbcb36eb9463e7c5a8311/.github/workflows/deploy_doc.yml#L36).

Once it's all signed off, I'm thinking of creating a PR to `dev` specifying the GitHub Action to take place upon push to `dev` branch on [line 6 of deploy_doc.yml](https://github.com/ericpien/yfinance/blob/40e93881cfd274871f3dbcb36eb9463e7c5a8311/.github/workflows/deploy_doc.yml#L6) so we can observe it get built and published once merged. If you have another branch for me to test on, I can do that instead. Then, submit a patch PR so documentation update takes place only upon push to `main` branch going forward.

**Reply by ValueRaider** [reply]
`documentation` now rebased so you can start a PR.

I need to think about what `on -> push -> branches` should be long-term. `main` might make more sense so webpages always match `main` (which always match last release). Can it be `documentation` for now?

**Reply by ericpien** [reply]
I can't push to `documentation` and also publish in the same branch's directory `docs` as this is [not allowed to keep the repo's other files safe](https://github.com/peaceiris/actions-gh-pages/issues/363). What I could do is push to another branch (i.e. `dev` or other test branch of preference), and have the workflow set up to build and publish the respective files to `documentation/docs` directory, from which the site will deploy. (i.e. in my fork, I have it build on push to [`docs` branch](https://github.com/ericpien/yfinance/blob/0ba753985f7ded62911e04c0f8aa36e1e66253a8/.github/workflows/deploy_doc.yml), publish to- and deploy from [`documentation` branch's](https://github.com/ericpien/yfinance/tree/documentation) `/docs` directory for testing purpose.) Given this, do you want me to target `dev` for the PR? 

Eventually, I agree push to `main` makes the most sense to keep the documentation on the most recent / stable release.

**Reply by ValueRaider** [reply]
I'm open-minded on this but here is my suggestion:
- target `dev-documented` first, branched from `dev`
- over next days/weeks: add docstrings, tweak rst files
- once unrelated changes in `dev` are pushed into a release, merge `dev-documented` into `dev` (or `main`)
